include/core/rtos/EventFlags.hpp

```diff
@@ -0,0 +1,122 @@
+
+#ifndef _EVT_RTOS_EVENTFLAGS_
+#define _EVT_RTOS_EVENTFLAGS_
+
+#include <core/rtos/Initializable.hpp>
+
+namespace core::rtos {
+
+/**
+ * Class that wraps a ThreadX Event Flags Group. Event flags are essentially
+ * a 32-bit mask, with each bit representing a different flag.\n\n
+ *
+ *
+ * This class extends Initializable, and like all other Initializable classes must be passed into the
+ * Threadx::startKernel() method as part of the initList parameter.
+ */
+class EventFlags : public Initializable {
+public:
+    /**
+     * Constructor for EventFlags.
+     *
+     * @param[in] name the name of the EventFlags.
+     */
+    EventFlags(char* name);
+
+    TXError init(core::rtos::BytePoolBase& pool) override;
+
+    /**
+     * Destructor for an EventFlags.
+     */
+    ~EventFlags();
+
+    /**
+     * Sets the eventFlags of the mask according to the setOption.
+     *
+     * @param[in] mask A bitmask describing what flags are to be interacted with.
+     * @param[in] clearNonMaskedFlags If true, all the flags not included in the mask will be set to 0.
+     * @return The first error found by the function (or Success if there was no error).
+     */
+    TXError set(uint32_t mask, bool clearNonMaskedFlags);
+
+    /**
+     * Gets the eventFlags of the mask according to the getOption.
+     *
+     * @param[in] mask A bitmask describing what flags are requested.
+     * @param[in] waitForAllFlags If the method should wait for EVERY flag in the bitmask
+     * to be true, or just ANY flag in the bitmask to be true.
+     * @param[in] clear If the method should clear outputted flags.
```

Comment:
I think outputted flags makes more sense. There's no definition of "output flags", but it is clear that if you set clear to true, then whatever flags you get back in your output will be set to 0 in the eventFlag.

Reply:
The argument here is that "outputted" is debatably not a word because "output" is the more common past tense of "output". This is entirely semantic, so you can ignore it if you like "outputted" better. It just sounds weird to me